modules/expo-bluesky-oauth-client/src/react-native-key.ts

```diff
@@ -0,0 +1,115 @@
+import {
+  jwkValidator,
+  Jwt,
+  JwtHeader,
+  jwtHeaderSchema,
+  JwtPayload,
+  jwtPayloadSchema,
+  Key,
+  VerifyOptions,
+  VerifyPayload,
+  VerifyResult,
+} from '@atproto/jwk'
+
+import {OauthClientReactNative} from './oauth-client-react-native'
+
+export class ReactNativeKey extends Key {
+  static async generate(kid: string, allowedAlgos: string[]) {
+    for (const algo of allowedAlgos) {
+      try {
+        // Note: OauthClientReactNative.generatePrivateJwk should throw if it
+        // doesn't support the algorithm.
+        const res = await OauthClientReactNative.generateJwk(algo)
+        const jwk = jwkValidator.parse({
+          ...res.privateKey,
+          key_ops: ['sign', 'verify'],
+          kid,
+        })
+        const use = jwk.use || 'sig'
+        return new ReactNativeKey(jwkValidator.parse({...jwk, use, kid}))
```

Comment:
note that `key_ops` and `use` are redundant and it's up to the implementation to decide which should be used. We settled for `use`.

```suggestion
        const res = await OauthClientReactNative.generateJwk(algo)
        const jwk = res.privateKey
        const use = jwk.use || 'sig'
        return new ReactNativeKey(jwkValidator.parse({...jwk, use, kid}))
```

So the code above should work. If you still get a validation error with that, it might be because of:
1) a bug in `jwkValidator`
2) the fact that the native module adds a `key_ops` that is not compatible with `use = "sig"`.

In case of #2, it would probably be best to remove `key_ops` in the native code, and make sure that the native code indeed creates signing keys.

Reply:
Yea, I ran into something there. It looks like the keys are capable of signing (tested the outputs in an external validator as well as ran `verifyJwt` on the output JWT). Let me work off what you have there and dig deeper to see what might be the problem 👍

Reply:
Wonderful, this works! Thanks!